client/src/redux/reducers/trip/tripSlice.js

```diff
@@ -0,0 +1,64 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { handleAsyncAction } from '../../handleAsync'
+import { REQUEST_STATE } from '../../states'
+import {
+  createTripAsync,
+  deleteTripAsync,
+  editTripAsync,
+  fetchTripsAsync,
+} from './thunks'
+/**
+ * @property {trips}: [{
+ *  _id: uuid, - Unique identifier for the trip
+ *  tripName: string, - Name of the trip, auto-generated
+ *  destination: string, - Destination of the trip (user input)
+ *  stagesPerDay: number, - Number of stages or activities planned per day (user input)
+ *  budget: number, - Total budget for the trip (user input)
+ *  numberOfDays: number - Total number of days for the trip (user input)
+ *  destinationLongitude: number - longitude (generated from BE)
+ *  destinationLatitude: number - latitude (generated from BE)
+ * }]
+ */
+export const tripSlice = createSlice({
+  name: 'trip',
+  initialState: {
+    trips: [],
+    status: REQUEST_STATE.IDLE,
+    error: null,
+  },
+  reducers: {},
+  extraReducers: (builder) => {
+    handleAsyncAction(builder, fetchTripsAsync, {
+      fulfilled: (state, action) => {
+        state.trips = action.payload
+      },
+    })
+    handleAsyncAction(builder, createTripAsync, {
+      fulfilled: (state, action) => {
+        state.trips.push(action.payload)
+      },
+    })
+    handleAsyncAction(builder, editTripAsync, {
+      fulfilled: (state, action) => {
+        const tripIndex = state.trips.findIndex(
+          (trip) => trip._id === action.payload._id,
+        )
+        if (tripIndex !== -1) {
+          state.trips[tripIndex] = {
```

Comment:
Hmmmmm should we be doing all these changes to our state.trips manually like this? Let's just do something like, where we call `editTripAsync`, change it to 

await dispatch(editTripAsync())
dispatch(fetchTripsAsync())

Reply:
it is perfectly fine to manually change the state, it keeps the slice up to date with the latest information. what you are suggesting with the dispatch is already what the app is doing. also adding an await in the first one doesnt guarantee that it finishes before fetching trip